File: src/main/java/com/github/theholychickn/theholychicknaddons/GoodMod.java
package com.github.theholychickn.theholychicknaddons;

import com.github.theholychickn.theholychicknaddons.owo.Awoo;
import com.github.theholychickn.theholychicknaddons.owo.Fuwwy;
import com.github.theholychickn.theholychicknaddons.owo.Meower;
import com.github.theholychickn.theholychicknaddons.owo.UwU;
import com.github.theholychickn.theholychicknaddons.owo.bwaa.Cute;
import com.github.theholychickn.theholychicknaddons.owo.bwaa.Furry;
import com.github.theholychickn.theholychicknaddons.owo.bwaa.OwO;
import com.github.theholychickn.theholychicknaddons.owo.bwaa.WoofWoof;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiScreen;
import net.minecraftforge.client.ClientCommandHandler;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@Mod(modid = "goodmod", useMetadata=true, clientSideOnly = true)
public class GoodMod {

    // Logger
    public static final Logger Kitten = LogManager.getLogger("good mod");

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event) {
        Meower.loadMeow();
        Awoo.init();
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        // Event subscriptions
        MinecraftForge.EVENT_BUS.register(this);
        MinecraftForge.EVENT_BUS.register(new Fuwwy());
        // Commands
        ClientCommandHandler.instance.registerCommand(new OwO());
        ClientCommandHandler.instance.registerCommand(new WoofWoof());
        ClientCommandHandler.instance.registerCommand(new Cute());
        ClientCommandHandler.instance.registerCommand(new Furry());
    }

    // Bonsai witchcraft
    @SubscribeEvent
    public void onTick(TickEvent.ClientTickEvent event) {
        if (event.phase != TickEvent.Phase.START) return;
        if (display != null) {
            Minecraft.getMinecraft().displayGuiScreen(display);
            display = null;
        }
    }

    static GuiScreen display = null;
    static public void setDisplay(GuiScreen newDisplay) {
        display = newDisplay;
    }
}

File: src/main/kotlin/com.github.theholychicken/GoodMod.kt
package com.github.theholychicken

import com.github.theholychicken.commands.*
import com.github.theholychicken.config.GuiConfig
import com.github.theholychicken.config.SellPricesConfig
import com.github.theholychicken.gui.ChestProfitTooltip
import com.github.theholychicken.gui.CroesusProfitHUD
import com.github.theholychicken.gui.MainCroesusHUD
import com.github.theholychicken.managers.*
import com.github.theholychicken.managers.apiclients.HttpClient
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.client.ClientCommandHandler
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLModDisabledEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import net.minecraftforge.fml.common.gameevent.TickEvent
import net.minecraftforge.fml.common.gameevent.TickEvent.ClientTickEvent
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

@Mod(modid = "goodmod", useMetadata = true, clientSideOnly = true)
class GoodMod {

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent?) {
        GuiConfig.loadConfig()
        if (GuiConfig.commandNames.isEmpty()) {
            GuiConfig.initConfig()
        }
        ItemDropParser.dropsConfig.loadConfig()
        if (ItemDropParser.dropsConfig.getList().isEmpty()) {
            ItemDropParser.initConfig()
        }
        SellPricesConfig.loadConfig()
        SellPricesConfig.absorbManualPrices() // for migration from legacy vers
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent?) {
        // Event subscriptions
        listOf(
            this,
            DungeonChestScanner,
            CroesusProfitHUD,
            MainCroesusHUD,
            ChestProfitTooltip
        ).forEach { MinecraftForge.EVENT_BUS.register(it) }

        // Commands
        listOf(
            OpenGuiCommand(),
            GetItemsCommand(),
            HelpCommand(),
            UpdateAuctionsCommand(),
            ConfigManualPricesTestCommand(), // test command
            KismetBoundsCommand()
        ).forEach { ClientCommandHandler.instance.registerCommand(it) }

        HttpClient.scheduleApiPulls()

        // Uncomment to access developer command - reloads AwA config when run
        // me, a few months later: wtf does this mean
        //ClientCommandHandler.instance.registerCommand(ReloadCommand());
    }

    @Mod.EventHandler
    fun onShutdown(event: FMLModDisabledEvent) {
        HttpClient.executor.shutdown()
    }

    // Bonsai witchcraft
    @SubscribeEvent
    fun onTick(event: ClientTickEvent) {
        if (event.phase != TickEvent.Phase.START || display == null) return
        mc.displayGuiScreen(display)
        display = null
    }

    companion object {
        var display: GuiScreen? = null
        val mc: Minecraft = Minecraft.getMinecraft()
        val logger: Logger = LogManager.getLogger("goodmod")

        var showUwU: Boolean = false
    }
}